crates/in_stream/src/tcp.rs

```diff
@@ -175,6 +179,7 @@ impl InStream<&mut [u8], &[u8]> for InStreamTcp {
         match stream.connect(addr) {
             Err(_) => Self::priv_new(
                 stream,
+                url.clone(),
                 Some(TcpConnectingData {
                     addr,
                     connect_timeout: match config.connect_timeout_ms {
```

Comment:
I know this was already written by someone else but we could get rid of the match and just do a config.connect_timeout_ms.map(|ms|{//add goes here})

Just reading through to gain better insights on how all this cones together

Reply:
good suggestion, will update :+1: